support/src/main/java/com/github/lizardev/xquery/saxon/support/trace/ExtensibleTraceExpression.java

```diff
@@ -14,7 +13,7 @@ public ExtensibleTraceExpression(Expression child) {
     }
 
     protected Expression getChild() {
-        return getFieldValue(this, "child");
+        return getChildExpression();
     }
 
     protected void setChild(Expression child) {
```

Comment:
if setChild causes perf issue then we can set this only when is different after simplify in `com.github.lizardev.xquery.saxon.support.trace.TraceExpressionComponent#simplify`